lib/connections/service.go

```diff
@@ -87,6 +88,8 @@ func NewService(cfg *config.Wrapper, myID protocol.DeviceID, mdl Model, tlsCfg *
 	}
 	cfg.Subscribe(service)
 
+	service.lans = determineLocalNetworkInCaseRateLimiterIsConfigured(cfg.Options())
```

Comment:
Sorry but this is becoming absurd.

```
rateLimitSet := opts.MaxRecvKbps > 0 || opts.MaxSendKbps > 0
if rateLimitSet && !opts.LimitBandwidthInLan {
  service.lans = getLans()
}
```

This is all starting to sound like https://github.com/EnterpriseQualityCoding/FizzBuzzEnterpriseEdition
I challenge you to find something this long in the standard library or any sensible go package.